src/v0/destinations/sfmc/transform.js

```diff
@@ -38,18 +45,24 @@ const getToken = async (clientId, clientSecret, subdomain) => {
     if (resp && resp.data) {
       return resp.data.access_token;
     }
-    throw new CustomError("Could not retrieve authorisation token", 400);
+    const status = resp.status || 400;
+    throw new NetworkError(
+      "Could not retrieve access token",
+      status,
+      {
+        [tags.TAG_NAMES.ERROR_TYPE]: getDynamicErrorType(status)
+      },
+      resp
+    );
   } catch (error) {
     if (!isEmpty(error.response)) {
-      throw new CustomError(
-        `Authorization Failed ${error.response.statusText}`,
-        error.response.status
+      throw new TransformationError(
+        `Authorization Failed ${error.response.statusText}`
       );
     } else {
       const httpError = nodeSysErrorToStatus(error.code);
-      throw new CustomError(
-        `Authorization Failed ${httpError.message}`,
-        httpError.status
+      throw new TransformationError(
+        `Authorization Failed ${httpError.message}`
```

Comment:
These are also network error as well

Reply:
Actually, this logic itself is incorrect. Only the Axios call should be inside the try-catch block. However, let's go with `NetworkError` for now.

Reply:
Yes

Reply:
updated